core/server/master/src/main/java/alluxio/master/file/meta/TtlBucketList.java

```diff
@@ -89,8 +89,19 @@ public void insert(Inode<?> inode) {
       // start time of this interval should be (ttlEndTimeMs / interval) * interval.
       long interval = TtlBucket.getTtlIntervalMs();
       bucket = new TtlBucket(interval == 0 ? ttlEndTimeMs : ttlEndTimeMs / interval * interval);
-      mBucketList.add(bucket);
+      synchronized (mBucketList) {
```

Comment:
@jsimsa  basically, creating new bucket and insert to list should be atomic, where having concurrent list won't help. @saltylin , this is a good catch! Thanks very much.